django/db/models/functions/base.py

```diff
@@ -25,6 +25,9 @@ def __init__(self, expression, output_field):
     def as_sql(self, compiler, connection, **extra_context):
         if 'db_type' not in extra_context:
             extra_context['db_type'] = self.output_field.db_type(connection)
+        # Support CharField casting without max_length.
```

Comment:
In my mind, it could be cleaner to add `Field.get_cast_db_type()` which returns `self.db_type()` and which `CharField` overrides. I think `Cast` should have minimal knowledge of model fields.

What do you think about putting the new property in the `DatabaseOperations`  class and calling it something like `cast_char_field_without_max_length`?